mousestyles/ultradian/__init__.py

```diff
@@ -0,0 +1,895 @@
+    of specified strain, mouse, feature and bin_width.
+    return the seasonal term and the plot of seasonal term
+    by mouse of a set of mouses in a strain
+
+    Parameters
+    ----------
+    strain: list, set or tuple
+        nonnegative integer indicating the strain number
+    mouse: list, set or tuple
+        nonnegative integer indicating the mouse number
+    feature: {"AS", "F", "M_AS", "M_IS", "W", "Distance"}
+        "AS": Active state probalibity
+        "F": Food consumed (g)
+        "M_AS": Movement outside homebase
+        "M_IS": Movement inside homebase
+        "W": Water consumed (g)
+        "Distance": Distance traveled
+    bin_width: int
+        number of minutes, the time interval for data aggregation
+    period_length: float or int
+        number of hours, usually the significant period
+        length indicated by Lomb-scargle model
+
+    Returns
+    -------
+    seasonal_plot: plot of seasonal term by mouse
+
+    Examples
+    --------
+    >>> res = plot_strain_seasonal(strains={0, 1, 2,}, mouse={0, 1, 2, 3},
+                                   feature="W",
+                                   bin_width=30, period_length = 24)
+    """
+
+    if (not all([isinstance(m, int)
+                 for m in mouse])) or (any([m < 0 for m in mouse])):
+        raise ValueError(
+            'Strain must be a non-negative integer')
+    if (not all([isinstance(m, int)
+                 for m in mouse])) or (any([m < 0 for m in mouse])):
+        raise ValueError(
+            'Mouse value must be a non-negative integer')
+    if feature not in ALL_FEATURES:
+        raise ValueError(
+            'Input value must in {"AS", "F", "M_AS", "M_IS", "W", "Distance"}')
+    if (not isinstance(bin_width, int)) or bin_width < 0 or bin_width > 1440:
+        raise ValueError(
+            'Bin width (minutes) must be a non-negative integer below 1440')
+    if period_length < 0:
+        raise ValueError(
+            'Peoriod length must be a non-negative float or int')
+
+    time = np.arange(0, period_length, bin_width / 60)
+    fig = plt.figure(figsize=(8, 8))
+    flag = 0
+
+    for strain in strains:
+        if flag == 0:
+            ax = fig.add_subplot(3, 1, strain + 1)
+            flag += 1
+        else:
+            ax = fig.add_subplot(3, 1, strain + 1, sharey=ax)
+        seasonal_all = strain_seasonal(strain, mouse, feature,
+                                       bin_width, period_length)
+        for i in np.arange(len(mouse)):
+            ax.plot(time, seasonal_all[i, :])
+        ax.legend(['mouse' + str(i) for i in np.arange(len(mouse))],
+                  loc='upper right', prop={'size': 10})
+        ax.set_title('strain ' + str(strain))
+        plt.xlabel('Time')
+        plt.ylabel('Seasonal Variation')
+    plt.suptitle(feature, fontsize=20)
+    fig.show()
+
+    return(fig)
+
+
+def mix_strain(data, feature, print_opt=True, nstrain=3, range=(3, 12), degree = 1):
```

Comment:
change input argument name, should not be range (conflict with existing function)